ember-primitives/src/components/accordion/accordion.gts

```diff
@@ -0,0 +1,150 @@
+import Component from '@glimmer/component';
+import { assert } from '@ember/debug';
+import { hash } from '@ember/helper';
+
+// temp
+//  https://github.com/tracked-tools/tracked-toolbox/issues/38
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+import { localCopy } from 'tracked-toolbox';
+
+import AccordionItem from './item';
+
+import type { WithBoundArgs } from '@glint/template';
+
+type AccordionSingleArgs = {
+  /**
+   * The type of accordion. If `single`, only one item can be selected at a time. If `multiple`, multiple items can be selected at a time.
+   */
+  type: 'single',
+  /**
+   * Whether the accordion is disabled.
+   */
+  disabled?: boolean;
+} & ({
+  /**
+   * The currently selected value. To be used in a controlled fashion in conjunction with `onValueChange`.
+   */
+  value: string;
+  /**
+   * A callback that is called when the selected value changes. To be used in a controlled fashion in conjunction with `value`.
+   */
+  onValueChange: (value: string | undefined) => void;
+  defaultValue?: never;
+} | {
+  value?: never;
+  onValueChange?: never;
+  /**
+   * The default value of the accordion. To be used in an uncontrolled fashion.
+   */
+  defaultValue?: string;
+});
+
+type AccordionMultipleArgs = {
+  /**
+   * The type of accordion. If `single`, only one item can be selected at a time. If `multiple`, multiple items can be selected at a time.
+   */
+  type: 'multiple'
+  /**
+   * Whether the accordion is disabled.
+   */
+  disabled?: boolean;
+} & ({
+  /**
+   * The currently selected values. To be used in a controlled fashion in conjunction with `onValueChange`.
+   */
+  value: string[];
+  /**
+   * A callback that is called when the selected values change. To be used in a controlled fashion in conjunction with `value`.
+   */
+  onValueChange: (value?: string[] | undefined) => void;
+  defaultValue?: never;
+} | {
+  value?: never;
+  onValueChange?: never;
+  /**
+   * The default values of the accordion. To be used in an uncontrolled fashion.
+   */
+  defaultValue?: string[];
+})
+
+export class Accordion extends Component<{
```

Comment:
Based on how many conditions are in here, I almost wonder if we should split (into private, non-exported components) this component into a `<SingleAccordion>` vs `<MultipleAccordion>` components -- I believe this would make testing easier as well?

what do you think?